src/crypto/scrypt.cpp

```diff
@@ -254,6 +254,7 @@ void scrypt_detect_sse2()
 
 void scrypt_1024_1_1_256(const char *input, char *output)
 {
-	char scratchpad[SCRYPT_SCRATCHPAD_SIZE];
+    static char scratchpad[SCRYPT_SCRATCHPAD_SIZE];
```

Comment:
Just chime in here. Do we really need `static` here? If we add `static` here, this function won't be thread safe anymore.
Although I didn't know ins/outs of scrypt algorithm, but I don't think `memset` would make difference here neither if `scrypt_1024_1_1_256_sp` would fully write to such char array.

What's the nature of the problem with musl for this segmentation fault? Possibly difference behavior as used by musl itself. Would you please add more info of such crash?